src/dune/dune_project.ml

```diff
@@ -830,21 +830,20 @@ let load_dune_project ~dir opam_packages =
 let load ~dir ~files ~infer_from_opam_files =
   let opam_packages =
     String.Set.fold files ~init:[] ~f:(fun fn acc ->
-        match Filename.split_extension fn with
-        | pkg, ".opam" when pkg <> "" ->
-          let name = Package.Name.of_string pkg in
-          let opam_file = Path.Source.relative dir fn in
-          let loc = Loc.in_file (Path.source opam_file) in
+        match Package.Name.of_basename fn with
+        | None -> acc
+        | Some name ->
```

Comment:
`".opam"` used to be filtered out. If I read the new code correctly, it is now considered as an empty package name

Reply:
Thanks for pointing this out. I'm still getting back to dune slowly :)

I suppose this is an issue everywhere where `Name.of_string` was used. So I moved the validation to `Name.of_string` and added a few functions to make sure the errors are good when the location is present.

Reply:
No problem :)

Changes look good